e2etest/local_transfer_test.go

```diff
@@ -595,3 +601,120 @@ func lenPendingActionMap(acts map[string][]action.SealedEnvelope) int {
 	}
 	return l
 }
+
+type chainCase1 struct {
+	cfgChainID       uint32
+	cfgJutlandHeight uint64
+	testChainID      uint32
+	err              error
+}
+
+func TestChainIDWithJutland(t *testing.T) {
+
+	testCase := []chainCase1{
+		{
+			1, 30, 1, nil, // tx chainID = node chainID, height < jutland
+		},
+		{
+			2, 31, 2, nil, // tx chainID = node chainID, height >= jutland
+		},
+		{
+			4, 31, 2, nil, // tx chainID != node chainID, height < jutland
+		},
+		{
+			2, 31, 4, nil, // tx chainID != node chainID, height < jutland
+		},
+		{
+			4, 30, 2, action.ErrChainID, // tx chainID != node chainID, height >= jutland
+		},
+		{
+			2, 30, 4, action.ErrChainID, // tx chainID != node chainID, height >= jutland
+		},
+		{
+			4, 10, 2, action.ErrChainID, // tx chainID != node chainID, height >= jutland
+		},
+		{
+			2, 10, 4, action.ErrChainID, // tx chainID != node chainID, height >= jutland
+		},
+	}
+
+	testFunc := func(c chainCase1) error {
+		ctx := context.Background()
+		cfg := config.Default
+		cfg.Genesis.BlockGasLimit = uint64(100000)
+		cfg.Chain.ID = c.cfgChainID
+		cfg.Genesis.JutlandBlockHeight = c.cfgJutlandHeight
+		registry := protocol.NewRegistry()
+		acc := account.NewProtocol(rewarding.DepositGas)
+		require.NoError(t, acc.Register(registry))
+
+		sf, err := factory.NewFactory(cfg, factory.InMemTrieOption(), factory.RegistryOption(registry))
+		if err != nil {
+			return err
+		}
+		ap, err := actpool.NewActPool(sf, cfg.ActPool)
+		if err != nil {
+			return err
+		}
+
+		blkMemDao := blockdao.NewBlockDAOInMemForTest([]blockdao.BlockIndexer{sf})
+		bc := blockchain.NewBlockchain(
+			cfg,
+			blkMemDao,
+			factory.NewMinter(sf, ap),
+		)
+		err = bc.Start(ctx)
+		if err != nil {
+			return err
+		}
+		for i := 0; i < 30; i++ {
+			tsf, err := action.NewTransfer(
+				uint64(i)+1,
+				big.NewInt(100),
+				identityset.Address(27).String(),
+				[]byte{}, uint64(100000),
+				big.NewInt(1).Mul(big.NewInt(int64(i)+10), big.NewInt(unit.Qev)),
+			)
+			if err != nil {
+				return err
+			}
+
+			bd := &action.EnvelopeBuilder{}
+			elp1 := bd.SetAction(tsf).
+				SetChainID(c.testChainID).
+				SetNonce(uint64(i) + 1).
+				SetGasLimit(100000).
+				SetGasPrice(big.NewInt(1).Mul(big.NewInt(int64(i)+10), big.NewInt(unit.Qev))).Build()
+			selp1, err := action.Sign(elp1, identityset.PrivateKey(0))
+			if err != nil {
+				return err
+			}
+
+			err = ap.Add(context.Background(), selp1)
+			if err != nil {
+				return err
+			}
+			blk, err := bc.MintNewBlock(testutil.TimestampNow())
+			if err != nil {
+				return err
+			}
+			err = bc.CommitBlock(blk)
+			if err != nil {
+				return err
+			}
+			err = bc.Stop(ctx)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+	for _, c := range testCase {
```

Comment:
don't need to start a chain for each testCase
just start one chain, create a transfer for each one of testCase (total 8)